projects/i18n/switch/language-switcher.ts

```diff
@@ -0,0 +1,53 @@
+import {Inject, Injectable, Optional} from '@angular/core';
+import {TuiLanguage, TuiLanguageName} from '@taiga-ui/i18n/interfaces';
+import {TUI_DEFAULT_LANGUAGE} from '@taiga-ui/i18n/tools';
+import {BehaviorSubject, Observable, of} from 'rxjs';
+
+import {TUI_LANGUAGE_CHUNK_LOADER} from './language-chunk-loader-token';
+import {tuiAsyncLoadLanguage} from './load-lang';
+import {
+    TUI_LANGUAGE_DEFAULT_STORAGE,
+    TUI_LANGUAGE_STORAGE_KEY,
+    TuiLangChunkLoader,
+} from './options';
+
+// @dynamic
+@Injectable({providedIn: `root`})
+export class TuiLanguageSwitcher extends BehaviorSubject<Observable<TuiLanguage>> {
+    constructor(
+        @Inject(TUI_DEFAULT_LANGUAGE)
+        private readonly fallback: TuiLanguage,
+        @Inject(TUI_LANGUAGE_STORAGE_KEY)
+        private readonly key: string,
+        @Inject(TUI_LANGUAGE_DEFAULT_STORAGE)
+        private readonly storage: Storage,
+        @Optional()
+        @Inject(TUI_LANGUAGE_CHUNK_LOADER)
+        private readonly loader: TuiLangChunkLoader | null,
+    ) {
+        super(tuiAsyncLoadLanguage({key, storage, fallback, loader}));
+    }
+
+    get currentLanguage(): TuiLanguageName {
+        return (this.storage.getItem(this.key) as TuiLanguageName) ?? this.fallback.name;
+    }
+
+    set(language: TuiLanguageName): void {
```

Comment:
It is `Subject`.
Let's rename this method to `next` to correctly implements [its interface](https://github.com/ReactiveX/rxjs/blob/9a85e01625a1ee7a23e4ab2c2c6d3089a58e2e54/src/internal/Subject.ts#L60) ?

Reply:
<img width="853" alt="image" src="https://user-images.githubusercontent.com/12021443/182602181-0f7781e8-11a1-42cb-8f5f-a81752a8ed14.png">

Reply:
<img width="403" alt="image" src="https://user-images.githubusercontent.com/12021443/182602230-f22553bf-e350-4aa9-b283-4221a8d1195b.png">